api/tests/certification/complementary-certification/unit/domain/usecases/attach-badges-to-complementary-certification_test.js

```diff
@@ -0,0 +1,355 @@
+    // eslint-disable-next-line mocha/no-setup-in-describe
+    [
+      { label: 'no badges found', resolve: [] },
+      { label: 'not all badges found', resolve: [{ badgeId: 1 }] },
+    ].forEach((assessment) => {
+      // eslint-disable-next-line mocha/no-setup-in-describe
+      context(`when  ${assessment.label}`, function () {
+        it('should throw a not found error', async function () {
+          // given
+          const complementaryCertificationRepository = {
+            getById: sinon.stub().withArgs(12).resolves({ id: 12 }),
+          };
+          const badgeRepository = {
+            findAllByIds: sinon.stub().withArgs(assessment.badges).resolves(assessment.resolve),
+          };
+
+          // when
+          const error = await catchErr(attachBadgesToComplementaryCertification)({
+            complementaryCertificationId: 12,
+            complementaryCertificationBadgesToAttachDTO: [
+              { badgeId: 1, level: 1 },
+              { badgeId: 2, level: 2 },
+            ],
+            complementaryCertificationRepository,
+            badgeRepository,
+          });
+
+          // then
+          expect(complementaryCertificationRepository.getById).to.have.been.calledOnce;
+          expect(error).to.be.instanceOf(NotFoundError);
+          expect(error.message).to.equal("Un ou plusieurs badges n'existent pas.");
+        });
+      });
+    });
+  });
+
+  context('when complementary certification does exist and levels checks are ok', function () {
+    context('when complementary certification badges are already attached to the profile', function () {
+      it('should detach old complementary certification badges', async function () {
+        // given
+        const domainTransaction = {
+          knexTransaction: Symbol('transaction'),
+        };
+        sinon.stub(DomainTransaction, 'execute').callsFake((callback) => {
+          return callback(domainTransaction);
+        });
+        const badge1 = domainBuilder.buildBadge({ id: 123 });
+        const badge2 = domainBuilder.buildBadge({ id: 456 });
+
+        const complementaryCertificationRepository = {
+          getById: sinon
+            .stub()
+            .withArgs(123)
+            .resolves(
+              domainBuilder.buildComplementaryCertification({
+                id: 123,
+              }),
+            ),
+        };
+        const badgeRepository = {
+          findAllByIds: sinon.stub().withArgs([123, 456]).resolves([badge1, badge2]),
+        };
+        const complementaryCertificationBadgesRepository = {
+          attach: sinon.stub().resolves(),
+          detachByIds: sinon.stub(),
+          getAllIdsByTargetProfileId: sinon.stub(),
+        };
+        const ComplementaryCertificationBadgeToAttachValidator = {
```

Comment:
Tests KO à cause de la majuscule un peu partout sur ComplementaryCertificationBadgeToAttachValidator

```suggestion
        const complementaryCertificationBadgeToAttachValidator = {
```

+ ligne 208